src/int_allocator.rs

```diff
@@ -29,62 +37,66 @@ impl IntAllocator {
     pub fn new() -> Self {
         let mut r = IntAllocator {
             u8_vec: Vec::new(),
-            node_vec: Vec::new(),
+            pair_vec: Vec::new(),
+            atom_vec: Vec::new(),
         };
         r.u8_vec.reserve(1024 * 1024);
+        r.atom_vec.reserve(256);
+        r.pair_vec.reserve(256);
         r.u8_vec.push(1_u8);
         // Preallocated empty list
-        r.node_vec
-            .push(NodePtr::Atom(IntAtomBuf { start: 0, end: 0 }));
+        r.atom_vec.push(IntAtomBuf { start: 0, end: 0 });
```

Comment:
Are these lists pointers to growable memory?

Reply:
they aren't pointers, they're indices. And yes, the vectors they point into can grow.

Reply:
indices have two main advantages:

1. the memory can grow without having to update pointers
2. we can use 32 bit integers instead of full 64 bit pointers, decreasing memory pressure